src/main/java/com/adobe/aem/compgenerator/utils/DialogUtils.java

```diff
@@ -298,6 +348,45 @@ private static Node updateDefaultNodeStructure(Document document, Element root)
                 .appendChild(createUnStructuredNode(document, "items"));
     }
 
+    /**
+     * Creates the default tab node structure.
+     *
+     * @param document The {@link Document} object
+     * @param root The {@link Element} object
+     * @return the node
+     */
+    private static Node createTabsParentNodeStructure(Document document, Element root) {
+        Element containerElement = createNode(document, "content", Constants.RESOURCE_TYPE_CONTAINER);
+        Node containerNode = root.appendChild(containerElement);
+        Element tabsElement = createNode(document, "tabs", Constants.RESOURCE_TYPE_TABS);
+
+        return containerNode.appendChild(createUnStructuredNode(document, "items")).appendChild(tabsElement)
+                .appendChild(createUnStructuredNode(document, "items"));
+    }
+
+    /**
+     * Creates the tab structure.
+     *
+     * @param document The {@link Document} object
+     * @param tab the tab
+     * @param node The {@link Node} object
+     * @return the node {@link Node} object
+     */
+    private static Node createTabStructure(Document document, Tab tab, Node node) {
+        Element tabElement = createNode(document, tab.getId(), Constants.RESOURCE_TYPE_CONTAINER);
+        tabElement.setAttribute(Constants.PROPERTY_JCR_TITLE, tab.getLabel());
```

Comment:
Since `tab.isValid()` does not ensure that the label is non-blank, we need to either update `tab.isValid()` to do so *or* we need to default this value to a capitalized version of `tab.getId()` (e.g. `CaseUtils.toCamelCase(tab.getId(), true)`) if label is blank.

Reply:
Agreed. Updated to use the ID as per your suggestions.